src/cli/commands/fellowship/joinGame.ts

```diff
@@ -0,0 +1,118 @@
+export const joinGame = new Command("joinGame")
+  .description("Join an existing game in a Rankify instance")
+  .argument("<instance>", "Address or instanceId of the Rankify instance")
+  .argument("<gameId>", "ID of the game to join")
+  .option("-r, --rpc <url>", "RPC endpoint URL. If not provided, RPC_URL environment variable will be used")
+  .option(
+    "-k, --key <privateKey>",
+    "Private key or index to derive from mnemonic for signing transactions. If not provided, PRIVATE_KEY environment variable will be used"
+  )
+  .action(async (instanceAddress, gameId, options) => {
+    const spinner = ora("Initializing clients...").start();
+
+    try {
+      const publicClient = await createPublic(options.rpc);
+      const walletClient = await createWallet(options.rpc, resolvePk(options.key, spinner));
+      const chainId = Number(await publicClient.getChainId());
+
+      const resolvedInstanceAddress = await CLIUtils.resolveInstanceAddress(
+        instanceAddress,
+        chainId,
+        publicClient,
+        spinner
+      );
+
+      spinner.text = "Creating Rankify player client...";
+      const account = walletClient.account?.address;
+
+      if (!account) {
+        spinner.fail("No account available");
+        throw new Error("No account available");
+      }
+
+      const player = new RankifyPlayer({
+        publicClient,
+        walletClient,
+        chainId,
+        instanceAddress: resolvedInstanceAddress,
+        account,
+      });
+
+      const gmWalletClient = await createWallet(options.rpc);
+
+      const gameMaster = new GameMaster({
+        walletClient: gmWalletClient,
+        publicClient,
+        chainId,
+      });
+
+      spinner.text = "Getting public key...";
+
+      const privateKeyBytes = hexToBytes(resolvePk(options.key, spinner) as Hex);
+      const publicKeyBytes = secp256k1.getPublicKey(privateKeyBytes, false);
+      const publicKey = bytesToHex(publicKeyBytes) as Hex;
+      const gameIdBigInt = BigInt(gameId);
+
+      spinner.text = "Validating game join eligibility...";
+      const { result: isValid, errorMessage } = await gameMaster.validateJoinGame({
+        instanceAddress: resolvedInstanceAddress,
+        gameId: gameIdBigInt,
+        participant: account,
+      });
+
+      if (!isValid) {
+        spinner.fail(`Cannot join game: ${errorMessage}`);
+        process.exit(1);
+      }
+
+      spinner.text = "Getting attestation from game master...";
+      const blockTimestamp = await publicClient.getBlock({ blockTag: "latest" }).then((block) => block.timestamp);
+      const timeToJoin = Number(blockTimestamp) + 60 * 10; // 10 minutes to join
+
+      console.log({
+        instanceAddress: resolvedInstanceAddress,
+        gameId: gameIdBigInt,
+        participant: account,
+        participantPubKeyHash: keccak256(encodePacked(["string"], [publicKey])),
+      });
+
+      const { signature: gmSignature, gmCommitment, deadline } = await gameMaster.signJoiningGame(
```

Comment:
Either signature should be passed in to this CLI, or GM key should be passed as argument